gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -24,4 +30,548 @@
+      return new MachineTypeListOption(ComputeRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to specify the machine type's fields to be returned by the RPC call. If
+     * this option is not provided all machine type's fields are returned.
+     * {@code MachineTypeListOption.fields} can be used to specify only the fields of interest.
+     * {@link MachineType#machineTypeId()} is always returned, even if not specified.
+     */
+    public static MachineTypeListOption fields(MachineTypeField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("items(").append(MachineTypeField.selector(fields)).append("),nextPageToken");
+      return new MachineTypeListOption(ComputeRpc.Option.FIELDS, builder.toString());
+    }
+  }
+
+  /**
+   * Class for specifying region get options.
+   */
+  class RegionOption extends Option {
+
+    private static final long serialVersionUID = 2025084807788610826L;
+
+    private RegionOption(ComputeRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the region's fields to be returned by the RPC call. If this
+     * option is not provided all region's fields are returned. {@code RegionOption.fields} can be
+     * used to specify only the fields of interest. {@link Region#regionId()} is always
+     * returned, even if not specified.
+     */
+    public static RegionOption fields(RegionField... fields) {
+      return new RegionOption(ComputeRpc.Option.FIELDS, RegionField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying region list options.
+   */
+  class RegionListOption extends Option {
+
+    private static final long serialVersionUID = 3348089279267170211L;
+
+    private RegionListOption(ComputeRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the maximum number of regions to be returned.
+     */
+    public static RegionListOption maxResults(long maxResults) {
+      return new RegionListOption(ComputeRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /**
+     * Returns an option to specify the page token from which to start listing regions.
+     */
+    public static RegionListOption startPageToken(String pageToken) {
+      return new RegionListOption(ComputeRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to specify the region's fields to be returned by the RPC call. If this
+     * option is not provided all region's fields are returned. {@code RegionListOption.fields} can
+     * be used to specify only the fields of interest. {@link Region#regionId()} is always
+     * returned, even if not specified.
+     */
+    public static RegionListOption fields(RegionField... fields) {
+      StringBuilder builder = new StringBuilder();
+      builder.append("items(").append(RegionField.selector(fields)).append("),nextPageToken");
```

Comment:
With the "Selectable" interface we could also make fields -> list string a common method...